src/amo/containers/App.js

```diff
@@ -1,46 +1,65 @@
-/* global navigator, window */
+/* @flow */
+/* eslint-disable react/sort-comp */
+/* global $PropertyType, Event, Navigator, Node, navigator, window */
 import config from 'config';
-import React, { PropTypes } from 'react';
+import { oneLine } from 'common-tags';
+import React from 'react';
 import cookie from 'react-cookie';
 import Helmet from 'react-helmet';
 import { connect } from 'react-redux';
 import LoadingBar from 'react-redux-loading-bar';
 import { compose } from 'redux';
 
 import SearchForm from 'amo/components/SearchForm';
-import { getErrorComponent } from 'amo/utils';
+import { getDjangoBase62, getErrorComponent } from 'amo/utils';
 import Footer from 'amo/components/Footer';
 import MastHead from 'amo/components/MastHead';
 import { addChangeListeners } from 'core/addonManager';
-import { setUserAgent as setUserAgentAction } from 'core/actions';
+import {
+  logOutUser as logOutUserAction, setUserAgent as setUserAgentAction,
+} from 'core/actions';
 import { INSTALL_STATE } from 'core/constants';
 import DefaultErrorPage from 'core/components/ErrorPage';
 import InfoDialog from 'core/containers/InfoDialog';
 import translate from 'core/i18n/translate';
 import log from 'core/logger';
+import type { UrlFormatParams } from 'core/api';
+import type { ApiStateType } from 'core/reducers/api';
+import type { DispatchFunc } from 'core/types/redux';
+import type { InstalledAddon } from 'core/reducers/installations';
 
 import 'amo/css/App.scss';
 import 'core/fonts/fira.scss';
 
+interface MozNavigator extends Navigator {
+  mozAddonManager?: Object,
+}
+
+type AppProps = {
+  ErrorPage: typeof DefaultErrorPage,
+  FooterComponent: typeof Footer,
+  InfoDialogComponent: typeof InfoDialog,
+  MastHeadComponent: typeof MastHead,
+  _addChangeListeners: () => void,
+  _navigator: typeof navigator,
+  authToken?: string,
+  authTokenValidFor?: number,
+  children: any,
+  clientApp: string,
+  handleGlobalEvent: () => void,
+  i18n: Object,
+  lang: string,
+  location: UrlFormatParams,
+  logOutUser: () => void,
+  mozAddonManager: $PropertyType<MozNavigator, 'mozAddonManager'>,
+  setUserAgent: () => void,
+  userAgent: string,
+}
 
 export class AppBase extends React.Component {
-  static propTypes = {
-    ErrorPage: PropTypes.node.isRequired,
-    FooterComponent: PropTypes.node.isRequired,
-    InfoDialogComponent: PropTypes.node.isRequired,
-    MastHeadComponent: PropTypes.node.isRequired,
-    _addChangeListeners: PropTypes.func,
-    _navigator: PropTypes.object,
-    children: PropTypes.node,
-    clientApp: PropTypes.string.isRequired,
-    handleGlobalEvent: PropTypes.func.isRequired,
-    i18n: PropTypes.object.isRequired,
-    lang: PropTypes.string.isRequired,
-    location: PropTypes.object.isRequired,
-    mozAddonManager: PropTypes.object,
-    setUserAgent: PropTypes.func.isRequired,
-    userAgent: PropTypes.string,
-  }
+  mastHead: Node;
```

Comment:
It's an annotation of a [class property](https://flow.org/en/docs/types/classes/#toc-class-fields-properties) which is cool because [things like refs](https://github.com/kumar303/addons-frontend/blob/556b2a8a890706aa50708e142f7c7522fbc67a67/src/amo/containers/App.js#L199) tend to be a bit obscure if you're seeing them for the first time in test code. With the Flow annotations, you can always look at the top of the class and see all the ref properties.